src/GraphBLAS-sharp.Backend/Common/ClArray.fs

```diff
@@ -318,7 +319,7 @@ module ClArray =
 
         let kernel = clContext.Compile map
 
-        fun (processor: MailboxProcessor<_>) (inputArray: ClArray<'a>) ->
+        fun (processor: MailboxProcessor<_>) flag (inputArray: ClArray<'a>) ->
 
             let result =
                 clContext.CreateClArrayWithFlag(flag, inputArray.Length)
```

Comment:
What do you thinh abaut the following design.
Instead of two fuctions `CreateClArray` (which accepts three optional allocation-related argumets), and `CreateClArrayWithSpecificAllocationMode` (which aimed to reduce parameters numbers by introbuceing a set of predefined flags compositions) we can modify CreateClArray such that it accepts an optional structure with three fields (actually flags). One can predefine a set of instances of this structure to define most important allocation modes. 

I think, it should be implemeted in Brahma, not in GraphBLAS#, but I want to discuss such design before implementation.